src/imandra_goals.ml

```diff
@@ -148,14 +153,20 @@ let close_goal ?hints g =
   in
   let hints =
     match hints with
-    | None -> g.hints |> CCOption.map (fun mk -> mk ())
-    | Some _ -> hints
+    | None ->
+      g.hints
+      |> CCOption.map (fun mk -> mk ())
+      |> CCOption.get_or ~default:Hints.default
+    | Some h -> h
   in
+
+  let hints = Hints.set_logic_config_ops g.logic_config_ops hints in
```

Comment:
If we can now do this in Imandra, I don't think there's any need to change anything in Imandra-goals? The caller can already provide their own `hints`.

Reply:
agreed! https://github.com/imandra-ai/imandra/pull/1529 should be enough

Reply:
alright!